Sources/Subprocess/AsyncBufferSequence.swift

```diff
@@ -214,14 +214,35 @@ extension AsyncBufferSequence {
                 }
 
                 while true {
+                    // Whether the buffer ends with a potential partial line ending sequence.
+                    // In this case we need to buffer more content until we have a complete line ending sequence so that peeking works properly.
+                    func bufferHasPotentialPartialLineSequence() -> Bool {
+                        guard Encoding.CodeUnit.self is UInt8.Type else {
+                            // Only the \r\n sequence requires multiple code units in the UTF-16 and UTF-32 encodings
+                            return self.buffer.last == carriageReturn
+                        }
+                        switch self.buffer.last {
+                        case carriageReturn, newLine1, lineSeparator1, paragraphSeparator1:
+                            return true
+                        default:
+                            switch self.buffer.suffix(2) {
+                            case [lineSeparator1, lineSeparator2], [paragraphSeparator1, paragraphSeparator2]:
+                                return true
+                            default:
+                                return false
+                            }
+                        }
+                    }
                     // Step 1: Load more buffer if needed
-                    if self.startIndex >= self.buffer.count {
+                    var partialLineSequence = bufferHasPotentialPartialLineSequence()
+                    while partialLineSequence || self.startIndex >= self.buffer.count {
```

Comment:
If the line sequence is just `...text...\r... text...\r` where there is never a `\n`, does this mean we'd just read in all buffered data at once (since `partialLineSequence` will keep being `true`)? 

Would it work to just unconditionally read one more byte to capture the potential `\n`, regardless of the set buffer size?